sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -416,6 +418,36 @@ private void refreshHasValueRestrictions(Map<Resource, URI> restrictions) throws
         }
     }
 
+    private void refreshAllValuesFromRestrictions(Map<Resource, URI> restrictions) throws QueryEvaluationException {
```

Comment:
I think it would be good to outline the flow of logic here.  E.g. refreshes allValuesRestrictions by creating a map of maps from the value class to a map that associates restrictions with properties.